lib/ramble/ramble/test/end_to_end/package_manager_provenance.py

```diff
@@ -59,14 +60,29 @@ def test_spack_package_manager_provenance_zlib(mock_applications, request):
     workspace("analyze", global_args=global_args)
 
     results_file = os.path.join(ws.root, "results.latest.txt")
-
     assert os.path.isfile(results_file)
 
     with open(results_file) as f:
         data = f.read()
         assert "Software definitions" in data
         assert "spack packages:" in data
-        assert "zlib" in data
+        assert "zlib @" in data
+
+    workspace("analyze", "-f", "json", global_args=global_args)
+
+    results_file = os.path.join(ws.root, "results.latest.json")
+    assert os.path.isfile(results_file)
+
+    with open(results_file) as f:
+        import json
+
+        data = json.load(f)
+        for data in data["experiments"]:
+            pkg_list = data["SOFTWARE"]["spack"]
+            names = [pkg["name"] for pkg in pkg_list]
+            assert "SOFTWARE" in data
+            assert "spack" in data["SOFTWARE"]
+            assert "zlib" in names
 
```

Comment:
If we're using an actual installation of zlib, I think something needs to happen to uninstall the environment that was installed here, otherwise we're messing with people's spack install directories just by running tests.

Reply:
To complicate things more... we probably only want to uninstall zlib if it wasn't installed before we ran `workspace setup`.

So, we might need some logic around making sure we clean this up properly.

Reply:
Hmm... something you might be able to do is add a config.yaml into the auxiliary software files that changes the install dir to be the same as the environment path...

Reply:
That's a really great idea! Updated